srml/contract/src/lib.rs

```diff
@@ -14,41 +14,64 @@
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module provides the ability to create smart-contract accounts and send them messages.
+//! A smart-contract is represented by an account with associated code and storage. 
+//! When a smart-contract account receives a message, the associated code gets executed.
+//! This code can alter the storage entries of the associated account, create new smart-contracts, or send messages to existing smart-contracts.
 //!
-//! This module provides an ability to create smart-contract accounts and send them messages.
-//! A smart-contract is an account with associated code and storage. When such an account receives a message,
-//! the code associated with that account gets executed.
+//! Senders must specify a gas limit with every call, as all instructions invoked by the smart-contract require gas.
+//! Unused gas is refunded after the transaction, regardless of the execution outcome. 
+//! If the gas limit is reached, then all changes made at the specific transaction level are reverted (including balance transfers).
+//! 
+//! **NOTE:** Transaction failures are typically not cascading. For example, if contract A calls B and B errors
+//! **NOTE:** Transaction failures are not always cascading. For example, if contract A calls contract B and B
+//! fails, A can decide how to handle that failure, either proceeding or reverting A's changes.
+//! 
+//! Finally, when the `staking` module determines an account is dead (i.e. account balance fell below the
+//! existential deposit), it reaps the account. This will delete the associated code and storage of the account.
+//! 
+//! ## Interface
+//! 
+//! ### Dispatchable functions ([`Call`])
+//!  
+//! * `PUT_CODE` - Stores the given binary Wasm code into the chains storage and returns its `codehash`.
+//! 
+//! * `call` - Makes a call to an account, optionally transferring some balance.
+//! 
+//! * `CREATE` - Deploys a new contract from the given `codehash`, optionally transferring some balance. This creates a new smart contract account and calls its contract deploy handler to initialize the contract.
 //!
-//! The code is allowed to alter the storage entries of the associated account,
-//! create smart-contracts or send messages to existing smart-contracts.
-//!
-//! For any actions invoked by the smart-contracts fee must be paid. The fee is paid in gas.
-//! Gas is bought upfront up to the, specified in transaction, limit. Any unused gas is refunded
-//! after the transaction (regardless of the execution outcome). If all gas is used,
-//! then changes made for the specific call or create are reverted (including balance transfers).
-//!
-//! Failures are typically not cascading. That, for example, means that if contract A calls B and B errors
-//! somehow, then A can decide if it should proceed or error.
-//!
-//! # Interaction with the system
```

Comment:
I find this (and what goes below) really useful from the standpoint of the user of this module (Although it went a bit dated). This is a bit related to [this comment](https://github.com/paritytech/substrate/pull/2045/files#r267347054) since it helps user to integrate this module with the rest of the system properly.

Reply:
commit issue. will be resolved!